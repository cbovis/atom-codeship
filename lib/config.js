'use babel';

import {
  CONFIG_KEY_POLLING_INTERVAL,
  CONFIG_KEY_CODESHIP_API_KEY,
  CONFIG_KEY_GITHUB_USERNAME,
} from './constants';

const config = {
  [CONFIG_KEY_CODESHIP_API_KEY]: {
    title: 'Codeship API Key',
    description:
      'Enter your codeship API key which can be found in the [account settings](https://app.codeship.com/user/edit).',
    type: 'string',
    default: '',
  },
  [CONFIG_KEY_GITHUB_USERNAME]: {
    title: 'GitHub Username',
    description: 'Enter your GitHub username so that we can identify your builds.',
    type: 'string',
    default: '',
  },
  [CONFIG_KEY_POLLING_INTERVAL]: {
    title: 'Polling Interval',
    description: 'Enter the interval (in seconds) on which to poll the build API.',
    type: 'integer',
    default: 15,
  },
};

const get = configKey => atom.config.get(`atom-codeship.${configKey}`);

export { config as schema };
export { get };
