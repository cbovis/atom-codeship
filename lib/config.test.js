import { schema } from './config';

describe('config/schema', () => {
  it('matches expected schema', () => {
    expect(schema).toMatchSnapshot();
  });
});
