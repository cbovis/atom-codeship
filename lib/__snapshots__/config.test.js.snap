// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config/schema matches expected schema 1`] = `
Object {
  "codeshipApiKey": Object {
    "default": "",
    "description": "Enter your codeship API key which can be found in the [account settings](https://app.codeship.com/user/edit).",
    "title": "Codeship API Key",
    "type": "string",
  },
  "githubUsername": Object {
    "default": "",
    "description": "Enter your GitHub username so that we can identify your builds.",
    "title": "GitHub Username",
    "type": "string",
  },
  "pollingInterval": Object {
    "default": 15,
    "description": "Enter the interval (in seconds) on which to poll the build API.",
    "title": "Polling Interval",
    "type": "integer",
  },
}
`;
